crates/chain/src/tx_data_traits.rs

```diff
@@ -5,21 +5,22 @@ use alloc::vec::Vec;
 
 /// Trait that "anchors" blockchain data to a specific block of height and hash.
 ///
-/// [`Anchor`] implementations must be [`Ord`] by the anchor block's [`BlockId`] first.
-///
-/// I.e. If transaction A is anchored in block B, then if block B is in the best chain, we can
+/// If transaction A is anchored in block B, then if block B is in the best chain, we can
 /// assume that transaction A is also confirmed in the best chain. This does not necessarily mean
 /// that transaction A is confirmed in block B. It could also mean transaction A is confirmed in a
 /// parent block of B.
 ///
+/// Every [`Anchor`] implementation must contain a [`BlockId`] parameter, and must implement
+/// [`Ord`]. When implementing [`Ord`], the first elements to compare must be the anchors'
```

Comment:
This is confusing.
What first elements mean?
You mean the `self.other` in [`std::cmp::Ord`](https://doc.rust-lang.org/std/cmp/trait.Ord.html)?

Maybe?
```suggestion
/// [`Ord`]. When implementing [`Ord`], the elements to compare must be the anchors'
```

Reply:
> What first elements mean?

That when you do the comparison in `Ord`, you must first compare the `BlockId`s, and then compare whatever you want if the blockids are the same. How would you rephrase it?

Reply:
Oh I see. Then maybe?
```suggestion
/// [`Ord`]. When implementing [`Ord`], the anchors' [`BlockId`]s should take precedence over other elements inside the [`Anchor`]s for comparison purposes.
```

Something like this. Feel free to rephrase it.

Reply:
Love it, thanks!